### PR TITLE
build: update deps to Nx 15.6.3, nitropack 1.0.0, astro 2.0.2

### PR DESCRIPTION
**Pull request by himyjan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-angular-plugin
- [ ] astro-angular
- [x] create-analog
- [ ] router
- [x] platform
- [ ] content

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
update blog-app to React 18.2.0
update Nx 15.6.2, typescript 4.9.4, nitropack 1.0.0 (support node 19), astro 2.0.1, cypress 12.4.0
update create-analog template-angular-v14, template-angular-v15 package version

nitropack 0.6.0 1.0.0 working -> nitropack 2.0.0 2.1.0 2.1.1 api not working [http://localhost:3000/api/v1/products](http://localhost:3000/api/v1/products)
image

react 18 with docusaurus 2.3.0 build error will fix in 2.3.1 version
https://github.com/facebook/docusaurus/issues/8592

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/create-analog/template-angular-v14/package.json [hunk 1]
```diff
@@ -15,26 +15,26 @@
   },
   "dependencies": {
     "@analogjs/router": "latest",
-    "@angular/animations": "^14.0.0",
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v14/package.json:18]: Revert these changes. Its ok to let the package manager determine the latest version to be installed

File: packages/create-analog/template-angular-v15/package.json [hunk 2]
```diff
@@ -15,29 +15,29 @@
...
-    "@angular/forms": "^15.0.0",
-    "@angular/platform-browser": "^15.0.0",
-    "@angular/platform-browser-dynamic": "^15.0.0",
-    "@angular/platform-server": "^15.0.0",
-    "@angular/router": "^15.0.0",
-    "@nrwl/angular": "~15.4.0",
-    "rxjs": "~7.5.6",
-    "tslib": "^2.4.0",
-    "zone.js": "~0.11.8"
+    "@angular/animations": "^15.1.1",
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v15/package.json:18]: Revert these changes

File: package.json [hunk 3]
```diff
@@ -46,72 +46,72 @@
...
     "@angular/router": "15.1.0",
-    "@nrwl/angular": "15.4.5",
-    "marked": "^4.2.4",
+    "@nrwl/angular": "15.6.2",
+    "marked": "^4.2.12",
     "front-matter": "^4.0.2",
-    "rxjs": "7.5.7",
-    "tslib": "^2.4.0",
-    "zone.js": "~0.11.8"
+    "rxjs": "7.8.0",
```
Comment (brandonroberts) [inline package.json:52]: In general, I think its safer to only update the packages Nx updates. It also helps keep compatibility with Angular release versions. Revert the `rxjs, tslib, and zone.js` changes

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: packages/create-analog/template-angular-v14/package.json [hunk 4]
```diff
@@ -29,12 +29,12 @@
   },
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v14/package.json:29]: Revert these changes

File: packages/create-analog/template-angular-v15/package.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
     "@angular/platform-browser-dynamic": "^15.0.0",
     "@angular/platform-server": "^15.0.0",
     "@angular/router": "^15.0.0",
-    "@nrwl/angular": "~15.4.0",
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v15/package.json:27]: Revert these changes

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by brandonroberts — APPROVED** [review]

File: apps/docs-app/package.json [hunk 6]
```diff
@@ -11,16 +11,30 @@
...
-    "@docusaurus/preset-classic": "2.0.1",
+    "@docusaurus/core": "2.2.0",
+    "@docusaurus/preset-classic": "2.2.0",
     "@mdx-js/react": "1.6.22",
     "clsx": "1.2.1",
     "prism-react-renderer": "1.3.5",
-    "react": "17.0.2",
-    "react-dom": "17.0.2"
+    "react": "18.2.0",
+    "react-dom": "18.2.0"
```
Comment (slorber) [inline apps/docs-app/package.json:20]: quick question but why do you want to use React 18 when Docusauurs does not provide yet official support?

You could as well stay on React 17 and use Docusaurus 2.3 (the recommended setup)

Comment (brandonroberts) [reply apps/docs-app/package.json:20]: Thanks, I didn't know it wasn't fully supported yet. I'm fine with going back to the recommended setup for now
